core/src/main/scala/io/scalac/mesmer/core/typeclasses/Encode.scala

```diff
@@ -0,0 +1,5 @@
+package io.scalac.mesmer.core.typeclasses
+
+trait Encode[I, O] {
```

Comment:
I think this is alright. The trait shows clear intent which is nice. I see no problem if we use it for different types if necessary. Note that encoding could be extended later to Encode[I, Either[EncodingError, O]]

Reply:
Sure, but we still don't make use of the `I` part - we always end up using `OtelTypeInstrumentation`

Reply:
I don't see the problem with being generic here even if for now we use it only for instrumentation. It doesn't hurt us when we have options for the future. It can hurt us if we are too specific. 

In the future, when somebody will try to do encoding, that person should reach for the Encoder trait and have a similar approach to what we just did. They will get encouraged if the type is generic. Contrary, if we use a specific type, people will get encouraged to use different approaches for the same thing (they won't see the similarity, because it's not generic but specific). 

If we find universal properties (like error handling, as in the example above) the universal type can encapsulate them. 

Also: other implications of Law of Demeter (as encoding itself is in no way coupled with Instrumentation). It generally feels good to me. :)